lib/file-handling/file-demuxer.js

```diff
@@ -24,21 +25,32 @@ function FileDemuxer(filePath, options) {
     return new FileDemuxer(filePath, options);
   }
 
-  assert(
-    utils.existsSync(filePath),
-    'File does not exist at the supplied path'
-  );
-
   options = merge(Object.create(FileDemuxer.DEFAULTS), options);
 
-  this._fileSize = fs.statSync(filePath).size;
+  //turn filePath into a readable Stream
+  if (typeof filePath === 'string') {
+    assert(
+      utils.existsSync(filePath),
+      'File does not exist at the supplied path'
+    );
+    options.fileSize = fs.statSync(filePath).size;
+    filePath = fs.createReadStream(filePath).on('error',
+      this.emit.bind(this, 'error'));
+  }
+
+  // TODO: pass size into constructor for file
+  // TODO for this._source pass in readableStream from blob store
```

Comment:
Both of these `TODO`s have been done. Remove before merge. `LONG TERM TODO` can be removed in favor of opening an issue.